File: api/mapping/post.py
from models import CreatePost, Post
from shared import slap
from . import set_comment

def create_post(create_post: CreatePost): 
    postKey = slap.generateUUID()
    post = Post(postKey, create_post.title, create_post.description, create_post.date,
                create_post.ownerUuid, create_post.topicKey)
    return {"_id": postKey, **vars(post)}, post

def set_post(post):
    comments = []
    if post["comments"] != []:
        for comment in post.comment:
            mapped_comment = set_comment(comment) 
            comments.append(mapped_comment)
        
    post = {
        "postKey": post["postKey"],
        "title": post["title"],
        "description": post["description"],
        "ownerUuid": post["ownerUuid"],
        "date": post["date"],
        "topicKey": post["topicKey"],
        "comments": comments
    }
    
    return post

File: mobile/models/post.js
export class CreatePost
{
    constructor(title, description, topicKey, ownerUuid)
    {
        this.title = title
        this.description = description
        this.topicKey = topicKey
        this.ownerUuid = ownerUuid
    }

    postJson(){
        const requestDate = new Date()
        return {
                "title": this.title,
                "description": this.description,
                "date": requestDate,
                "ownerUuid": this.ownerUuid,
                "topicKey": this.topicKey
              }


    }
}

export class Post {
    constructor(title, description, author, topic, date)
    {
        this.title = title
        this.description = description
        this.author = author
        this.topic = topic
        this.topic = date
    }
}